Lombiq.Hosting.Tenants.EmailQuotaManagement.Tests.UI/Extensions/TestCaseUITestContextExtensions.cs

```diff
@@ -21,26 +24,56 @@ public static class TestCaseUITestContextExtensions
     {
         await context.SignInDirectlyAndGoToDashboardAsync();
 
-        context.Missing(By.XPath(DashboardWarning));
+        context.Missing(By.XPath(DashboardExceededMessage));
 
         await context.GoToAdminRelativeUrlAsync("/Settings/email");
 
         CheckEmailsSentWarningMessage(context, exists: moduleShouldInterfere, maximumEmailQuota, 0);
 
-        await SendTestEmailAsync(context, SuccessfulSubject);
-        context.SuccessMessageExists();
+        var warningEmails = new List<int>();
+        for (int i = 0; i < maximumEmailQuota; i++)
+        {
+            await SendTestEmailAsync(context, SuccessfulSubject);
+            context.SuccessMessageExists();
+            CheckEmailsSentWarningMessage(context, exists: moduleShouldInterfere, maximumEmailQuota, i + 1);
+            var warningLevel = Convert.ToInt32((double)(i + 1) / maximumEmailQuota * 100) / 10 * 10;
+            if (!moduleShouldInterfere) continue;
 
-        CheckEmailsSentWarningMessage(context, exists: moduleShouldInterfere, maximumEmailQuota, 1);
-
-        await context.GoToDashboardAsync();
-        context.CheckExistence(By.XPath(DashboardWarning), exists: moduleShouldInterfere);
+            if (warningLevel >= 100)
+            {
+                await context.GoToDashboardAsync();
+                context.CheckExistence(By.XPath(DashboardExceededMessage), exists: moduleShouldInterfere);
+            }
+            else if (warningLevel >= 80)
+            {
+                await context.GoToDashboardAsync();
+                context.CheckExistence(
+                    By.XPath($"//p[contains(@class,'alert-warning')]" +
+                        $"[contains(.,'It seems that your site sent out {warningLevel.ToTechnicalString()}% of e-mail')]"),
+                    exists: moduleShouldInterfere);
+                if (!warningEmails.Contains(warningLevel))
+                {
+                    warningEmails.Add(warningLevel);
+                }
+            }
```

Comment:
While displaying the precise percentage above 80% is fine, I don't see how this tests that the module should only send a warning at 80% and 90% (in addition to at 100%). It shouldn't send a warning every time above 80%.

Reply:
Let's clear something because I'm not sure if we are on the same page. 

What's the goal with the admin dashboard warning message? Should it show a precise percentage above 80% or not? Because currently it doesn't, it only show 80% until it doesn't reach 90%, then if the quota is exceeded the danger message is show instead of the warning.
It's easier to show the exact percentage, then we don't need these roundings at all.

Reply:
My initial idea was to display just that you're above 80, or 90%, without the precise current percentage (though that would need slightly different wording in EmailQuotaError). I misunderstood this code, thinking it checks that. But yeah, having the current percentage displayed (rounded to the nearest full percentage) is better.